packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -653,6 +657,16 @@ export default class TradeStore extends BaseStore {
         const prev_proposal_info      = ObjectUtils.getPropertyValue(this.proposal_info, contract_type) || {};
         const obj_prev_contract_basis = ObjectUtils.getPropertyValue(prev_proposal_info, 'obj_contract_basis') || {};
 
+        // Sometimes the API doesn't forget old 'proposal' response and returns them with new 'proposal' response, so here
+        // we need to send 'forget' req for old proposal subscriptions and store the latest proposal req_id
+        if (this.proposal_req_id[contract_type] < response.echo_req.req_id) {
+            // if an old proposal subscription exist, send 'forget'
+            if (this.proposal_info[contract_type] && this.proposal_info[contract_type].id) {
+                WS.forget(this.proposal_info[contract_type].id);
+            }
+            this.proposal_req_id[contract_type] = response.echo_req.req_id;
```

Comment:
shouldn't we return after this to ignore this response since it's old?

Reply:
the line below will update the `proposal_info` obj with the most recent ones